src/scp/SCP.cpp

```diff
@@ -281,12 +281,20 @@ SCP::getLatestMessage(NodeID const& id)
 }
 
 bool
-SCP::isLatestStatement(SCPStatement const& st)
+SCP::shouldReplaceSt(SCPStatement const& st, SCPStatement const& hint)
 {
+    bool sameSlot = st.slotIndex == hint.slotIndex;
+    bool sameNodeID = st.nodeID == hint.nodeID;
+
+    if (!sameSlot || !sameNodeID)
```

Comment:
you probably don't want to use separate variables here: I don't trust compilers to lazy evaluate the second `==` in this case (that happens to be expensive)